prac_06/guitars.py

```diff
@@ -0,0 +1,29 @@
+from prac_06.guitar import Guitar
+
+
+def main():
+    """Get information about users guitars and print information."""
+    guitars = []
+    # guitars.append(Guitar("Gibson L-5 CES", 1922, 16035.40))
+    # guitars.append(Guitar("Line 6 JTV-59", 2010, 1512.9))
+
+    print("My guitars!")
+    name = input("Name: ")
+    while name != "":
+        year = int(input("Year: "))
+        cost = float(input("Cost: "))
+        guitars.append(Guitar(name, year, cost))
+        print()
+        name = input("Name: ")
+
+    print("These are my guitars: ")
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = ""
+        if guitar.is_vintage():
+            vintage_string = "(vintage)"
+        print(f"Guitar {i}: {guitar.name:>20} ({guitar.year:>4}), worth ${guitar.cost:10,.2f} {vintage_string}")
+        #  I tried multiple attempts to get the guitar name spacing working using
+        #  list comprehensions but couldn't figure it out in the end as I kept getting errors with generator expressions
+        #  max_name_length = max(len(guitar.name for guitar in guitars))
```

Comment:
I think python doesn't like this line because it is coming from the class. You can get the length of the guitar name by adding a len function to the class e.g.
"""
    def __len__(self):
        return self._data_len
"""
Then in the __init__ you would need """self.data_len = len(name)"""
And you can call it just like you called the is_vintage function in your loop. 
There is probably a more concise way to do it but that is one way I found.

Reply:
Ahh i see. That actually makes a lot of sense. Appreciate it.